synapse/lib/threads.py

```diff
@@ -226,26 +247,13 @@ def _run_work(self):
             if work is None:
                 return
 
-            self.fire('pool:work:init', work=work)
-
-            task, info = work
-
-            jid = info.get('jid')
             try:
 
-                func, args, kwargs = task
-                ret = func(*args, **kwargs)
-
-                # optionally generate a job event
-                if jid is not None:
-                    self.fire('job:done', jid=jid, ret=ret)
+                func, args, kwargs = work
+                func(*args, **kwargs)
 
             except Exception as e:
-
-                if jid is not None:
-                    self.fire('job:done', jid=jid, **s_common.excinfo(e))
-
-            self.fire('pool:work:fini', work=work)
+                logger.exception(e)
```

Comment:
``logger.exception('Error while executing task')`` is all that is needed here.  logger.exception pulls the exception data from the stack directly.

Reply:
ah, yea.  good call.  dunno where i got into the paradigm of logger.exception(Exception) but you're def right.